### ISSUE TITLE
Gateway hangs instead of returning 404

### ISSUE THREAD

**Issue by JustinDrake** [ipfs/go-ipfs#4157]
Both OB1 and Duo are running IPFS gateways for OpenBazaar, and fetching objects that are not found hangs the connection (instead of returning for example a 404).There doesn't seem to be a reasonable timeout. See example links below. 

https://gateway.ob1.io/ipfs/zdj7Wiy5Q5x1d6Ga9Ct3JAbWYDRqS7UPzvoxfmirF2zd2kPsb
https://gateway.duosear.ch/ipfs/zdj7Wiy5Q5x1d6Ga9Ct3JAbWYDRqS7UPzvoxfmirF2zd2kPsb

**Comment by hsanjuan** [comment]
@Stebalien is this something we want to address (custom timeout for gateway requests)?

Workaround is to run Nginx or any other reverse proxy middleware which gives control over this and many other settings, particularly for production deployments.

**Comment by Stebalien** [comment]
Yeah, probably. But it'll still likely take a few minutes before the gateway simply gives up.

**Comment by Stebalien** [comment]
IIRC, we already resolve files in two stages: resolve the root block then fetch the rest of the file. We'd want to add a timeout for the first part only.

**Comment by lidel** [comment]
- Kubo 0.38 will ship with #10950
